modules/apps/journal/journal-test/src/testIntegration/java/com/liferay/journal/subscriptions/test/JournalSubscriptionLocalizedContentTest.java

```diff
@@ -44,14 +82,112 @@ public class JournalSubscriptionLocalizedContentTest
+		super.setUp();
+
+		group.setName("Test Site in English", LocaleUtil.getDefault());
+		group.setName("Sitio de Pruebas en español", LocaleUtil.SPAIN);
+
+		group = _groupLocalService.updateGroup(group);
+	}
+
+	@Test
+	public void testSubscriptionLocalizedContentWithMacrosWhenAddingBaseModel()
+		throws Exception {
+
+		user = _userLocalService.updateLanguageId(
+			user.getUserId(), _language.getLanguageId(LocaleUtil.SPAIN));
+
+		setBaseModelSubscriptionBodyPreferences(
+			HashMapBuilder.put(
+				LocaleUtil.SPAIN, _EMAIL_ARTICLE_ADDED_BODY
+			).build(),
+			getSubscriptionAddedBodyPreferenceName());
+
+		addSubscriptionContainerModel(getDefaultContainerModelId());
+
+		_themeDisplay = _getThemeDisplay(creatorUser);
+
+		long journalArticlePrimaryKey = addBaseModel(
+			creatorUser.getUserId(), getDefaultContainerModelId());
+
+		Assert.assertEquals(1, MailServiceTestUtil.getInboxSize());
+
+		MailMessage mailMessage = MailServiceTestUtil.getLastMailMessage();
+
+		_themeDisplay.setUser(user);
+
+		Assert.assertEquals(
+			_getExpectedMailBody(
+				_EMAIL_ARTICLE_ADDED_BODY, journalArticlePrimaryKey),
+			mailMessage.getBody());
+	}
+
+	@Test
+	public void testSubscriptionLocalizedContentWithMacrosWhenUpdatingBaseModel()
+		throws Exception {
+
+		user = _userLocalService.updateLanguageId(
+			user.getUserId(), _language.getLanguageId(LocaleUtil.SPAIN));
+
+		setBaseModelSubscriptionBodyPreferences(
+			HashMapBuilder.put(
+				LocaleUtil.SPAIN, _EMAIL_ARTICLE_UPDATED_BODY
+			).build(),
+			getSubscriptionUpdatedBodyPreferenceName());
+
+		_themeDisplay = _getThemeDisplay(creatorUser);
+
+		long journalArticlePrimaryKey = addBaseModel(
+			creatorUser.getUserId(), getDefaultContainerModelId());
+
+		addSubscriptionContainerModel(getDefaultContainerModelId());
+
+		updateBaseModel(creatorUser.getUserId(), journalArticlePrimaryKey);
+
+		Assert.assertEquals(1, MailServiceTestUtil.getInboxSize());
+
+		MailMessage mailMessage = MailServiceTestUtil.getLastMailMessage();
+
+		_themeDisplay.setUser(user);
+
+		Assert.assertEquals(
+			_getExpectedMailBody(
+				_EMAIL_ARTICLE_UPDATED_BODY, journalArticlePrimaryKey),
+			mailMessage.getBody());
+	}
+
 	@Override
 	protected long addBaseModel(long userId, long containerModelId)
 		throws Exception {
 
+		ServiceContext serviceContext =
+			ServiceContextTestUtil.getServiceContext(
+				group.getGroupId(), userId);
+
+		if (_themeDisplay != null) {
+			serviceContext.setRequest(_themeDisplay.getRequest());
+		}
+
 		JournalArticle article = JournalTestUtil.addArticle(
-			userId, group.getGroupId(), containerModelId);
+			group.getGroupId(), containerModelId,
+			JournalArticleConstants.CLASS_NAME_ID_DEFAULT,
+			HashMapBuilder.put(
+				LocaleUtil.getDefault(), "Title"
+			).put(
+				LocaleUtil.SPAIN, "Título"
+			).build(),
+			null,
+			HashMapBuilder.put(
+				LocaleUtil.getDefault(), "Content in English"
+			).put(
+				LocaleUtil.SPAIN, "Contenido en español"
```

Comment:
@marcogalluzzi, there seems to be something off with special chars and the email server that we use, so I would suggest that we don't use special chars

Reply:
OK, I will avoid special characters. However, I don't know why HTML chars are escaped by the email server. This would avoid testing macro ARTICLE_DIFFS

Reply:
I've done this change in the test [dcc243b](https://github.com/liferay-content-management/liferay-portal/pull/5730/commits/dcc243bd23a77e215078332985cc4087025dc375). Please check if is OK because I'm not sure.

Reply:
Other solutions could be:

* Always set the `8bit` enconding in `MailEngine`. But I don't know if breaks some functionality.
* Set the "Content-Transfer-Encoding" header in the message when creating it in `JournalArticleLocalServiceImpl` by checking if the ServiceContext has it. This way I can pass this header from the test.